railties/lib/rails/generators/rails/app/templates/Gemfile.tt

```diff
@@ -3,6 +3,8 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby <%= "'#{RUBY_VERSION}'" -%>
 
+<%- if options[:database] == 'oracle' %>
+gem 'ruby-oci8', platforms: :ruby  # only for CRuby users<%- end -%>
```

Comment:
This should handling in `gem_for_database` method.
https://github.com/rails/rails/blob/782d28030730c14658a7cd66ee81e7f16c0a12cd/railties/lib/rails/generators/database.rb#L14-L28

@yahonda I'm curious, you changed this gem by #28257. Why didn't you keep `ruby-oci8` as CRuby dependency?

Reply:
One of the main reasons why I changed `ruby-oci8` to `activerecord-oracle_enhanced-adapter` is Oracle enhanced adapter supports both CRuby and JRuby. `ruby-oci8` can be installed only with CRuby.

If there is a way to add `ruby-oci8` as a dependent gem of Oracle enhanced adapter only for CRuby, I'd like to update Oracle enhanced adapter but I do not know how to.

Reply:
Thanks for your response.  
I thought that when generating a `Gemfile` in `rails new`, it would be good to add not only `activerecord-oracle_enhanced-adapter` but also `ruby-oci8`(only when using CRuby). 
If there is no problem that, I will suggest that to this PR.